core/model/src/main/java/org/eclipse/sw360/antenna/model/artifact/facts/ArtifactCoordinates.java

```diff
@@ -11,10 +11,36 @@
 
 package org.eclipse.sw360.antenna.model.artifact.facts;
 
+import com.github.packageurl.MalformedPackageURLException;
+import com.github.packageurl.PackageURL;
+import com.github.packageurl.PackageURLBuilder;
+
 public abstract class ArtifactCoordinates<T extends ArtifactCoordinates> implements ArtifactIdentifier<T> {
     public abstract String getName();
+
     public abstract String getVersion();
 
+    public abstract String getType();
+
+    protected PackageURLBuilder addPurlFacts(PackageURLBuilder builder) {
+        return builder;
+    }
+
+    public PackageURL getPurl() {
+        PackageURLBuilder builder =  PackageURLBuilder.aPackageURL()
+                .withType(getType())
+                .withName(getName());
+
+        if (getVersion() != null) {
```

Comment:
This explicit `null` check is still there, did you intentionally keep it?

Reply:
Version is according to the purl spec an optional value. I thought that it makes sense to not set the value, if the version is unknown instead of setting a null value, so yes this is intentional.

Reply:
My point is: Exactly that logic is already handled by the purl library (see the code I had linked in my original comment), so why also handle that case here?

Reply:
Ok accepted, actually, there was a major bug in the implementation, but now it is fixed